CONTRIBUTING.md

```diff
@@ -173,6 +173,62 @@ If you see Scala symbols unresolved (highlighted red) in IDEA please try the fol
 - Make sure there are no relevant poms in "File->Settings->Build Tools->Maven->Ignored Files"
 - Restart IDEA and click "Reload All Maven Projects" again
 
+#### Bloop
+
+[Bloop](https://scalacenter.github.io/bloop/) is a build server and a set of tools around Build Server Protocol (BSP)
+for Scala providing an integration path with IDEs that support it. In fact, you can generate a bloop project from
+Maven just for the Maven modules and profiles you are interested in it. E.g., in order to generate Bloop projects the
+Spark 3.2.0 dependency just for the production code run:
+```shell script
+mvn install ch.epfl.scala:maven-bloop_2.13:1.4.9:bloopInstall -pl aggregator -am \
+  -DdownloadSources=true \
+  -Dbuildver=320 \
+  -DskipTests \
+  -Dskip \
+  -Dmaven.javadoc.skip \
+  -Dmaven.scalastyle.skip=true \
+  -Dmaven.updateconfig.skip=true
+```
+
+With `--generate-bloop` we integrated Bloop project generation into `buildall`. It makes it easier to generate
+projects for multiple Spark dependencies using the same profiles as our regular build.
+It makes sure that the project files belonging to different Spark dependencies are
+not clobbered by repeated bloopInstall Maven plugin invocations and uses [jq](https://stedolan.github.io/jq/)
+to post-process JSON-formatted project files such that they compile project classes into non-overlapping
+set of output directories.
+
+You can now open the spark-rapids as a [BSP project in IDEA](https://www.jetbrains.com/help/idea/bsp-support.html)
+
+Another, and arguably more popular, use of Bloop arises in connection with
+[Scala Metals](https://scalameta.org/metals/) and [VS @Code](https://code.visualstudio.com/)
+
+# Bloop, Scala Metals, and Visual Studio Code
+
```

Comment:
it would be nice to have a brief description to this section as to what this is useful for.